tests/webTables.spec.ts

```diff
@@ -0,0 +1,202 @@
+      .getByRole("row", { name: "surgery" })
+      .getByRole("button", { name: "Edit" })
+      .click();
+
+    await expect(page.locator("h2")).toHaveText("Edit Specialty");
+
+    const specialtyNameInputField = page.getByRole("textbox");
+    await specialtyNameInputField.click();
+    await specialtyNameInputField.clear();
+    await specialtyNameInputField.fill("dermatology");
+    await page.getByRole("button", { name: "Update" }).click();
+
+    expect(page.getByRole("row", { name: "dermatology" }));
+
+    await page.getByText("Veterinarians").click();
+    await page.getByText("All").click();
+    await expect(targetRow.locator("td").nth(1)).toHaveText("dermatology");
+
+    await page.getByRole("link", { name: "Specialties" }).click();
+    await page
+      .getByRole("row", { name: "dermatology" })
+      .getByRole("button", { name: "Edit" })
+      .click();
+    await specialtyNameInputField.click();
+    await specialtyNameInputField.clear();
+    await specialtyNameInputField.fill("surgery");
+    await page.getByRole("button", { name: "Update" }).click();
+  });
+});
+
+test.describe("Web Tables-Specialties", async () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/");
+    await page.getByText("Specialties").click();
+  });
+
+  test("Validate specialty lists", async ({ page }) => {
+    await page.getByRole("button", { name: "Add" }).click();
+    await page.locator("#name").fill("oncology");
+    await page.getByRole("button", { name: "Save" }).click();
+
+    await page.waitForTimeout(1000);
+
+    const specialties = [];
+    const specialtyRows = page.locator("tbody tr input");
+
+    for (let specialty of await specialtyRows.all()) {
+      const specialtyName = await specialty.inputValue();
+      specialties.push(specialtyName);
+    }
+
+    expect(specialties).toEqual([
+      "radiology",
+      "surgery",
+      "dentistry",
+      "New specialty",
+      "oncology",
+    ]);
+
+    await page.getByText("Veterinarians").click();
+    await page.getByText("All").click();
+
+    const targetRow = page.getByRole("row", { name: "Sharon Jenkins" });
+    await targetRow.getByRole("button", { name: "Edit Vet" }).click();
+
+    await page.locator(".dropdown-arrow").click();
+
+    const allBoxes = await page
+      .locator('//input[@type="checkbox"]/following-sibling::label')
+      .allInnerTexts();
+
+    expect(allBoxes).toEqual(specialties);
+
+    await page.getByRole("checkbox", { name: "oncology" }).check();
+    await page.getByRole("checkbox", { name: "New specialty" }).uncheck();
+    await page.locator(".dropdown-arrow").click();
+    await page.getByRole("button", { name: "Save Vet" }).click();
+
+    await expect(targetRow.locator("td").nth(1)).toHaveText("oncology");
+
+    await page.getByRole("link", { name: "Specialties" }).click();
+
+    await page.getByRole("button", { name: "Delete" }).last().click();
+    await page.waitForTimeout(1000);
+
+    await page.goto("/");
```

Comment:
The goTo() method normally should not be used in the middle of the test. This is hot how user using the application.
Ask yourself, do you type `https://petclinic.bondaracademy.com` into the search bar of the browser after you delete the specialty to navigate to veterinarians? I think you didn't. So, your script should not do it as well.

Reply:
done